extra/import_data.py

```diff
@@ -1,197 +1,158 @@
+    if analysis is None:
+        logger.info('no new analysis in Alissa since last upload to VUSualizer')
+        sys.exit(0)
+    # after uploading all analyses to MongoDB, replace time of last upload with starttime of script (before calling Alissa)
+    db.replace_one({"version" : 1}, {"lastUpdatedOn" : start_time, "version" : 1}, True)
+
+
+def upload_to_mongodb(ONB01_analysis, accession_number, analyis_sources, patient_dn_no, VUS_response):
+    '''Function, for parsing data to the MongoDB'''
+    # connection with MongoDB and the correct database "vus" and collection "variant"
+    mongoclient = pymongo.MongoClient("mongodb://localhost:27017/")
+    db = mongoclient.vus.variant
+    
+    # if DNnr already in mongoDB, check if updated an replace if neccesary
+    if db.count_documents({"dn_no": patient_dn_no}) > 0:
+        logger.info('dn_no: %s already present' % patient_dn_no)
+        # get and parse "lastUpdatedOn" from MongoDB/Alissa for specific patient_dn_no
+        lastUpdatedOn_mongoDB = db.find_one({"dn_no" : patient_dn_no}, {"lastUpdatedOn":1, "_id":0})
+        for key, value in lastUpdatedOn_mongoDB.items():
+            lastUpdatedOn_mongoDB = value
+        lastUpdatedOn_Alissa = ONB01_analysis['lastUpdatedOn']
+        if lastUpdatedOn_Alissa == lastUpdatedOn_mongoDB:
+            logger.info('%s already in database, lastUpdatedOn Alissa is the same' % patient_dn_no) 
+            return
+        elif lastUpdatedOn_Alissa > lastUpdatedOn_mongoDB:
+            db.delete_many({"dn_no": patient_dn_no})
+            logger.info('removed: %s from database, start replacing with newer version' % patient_dn_no)
+        elif lastUpdatedOn_Alissa < lastUpdatedOn_mongoDB:
+            logger.info('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+            sys.exit('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+        else:
+            logger.info('unknown error lastUpdatedOn for patient %s, this should not happen' % patient_dn_no)
+            sys.exit('unknown error lastUpdatedOn for patient %s, this should not happen' % patient_dn_no)
+
+    # get all relevant info from one patient into one dictionary.
+    patient = {}
+    patient["dn_no"] = patient_dn_no
+    logger.info('start uploading to mongodb: %s' % patient_dn_no)
+    patient["patient_accession_no"] = accession_number
+    patient.update(ONB01_analysis)
+    patient["analysis"] = patient.pop("reference")
+    patient["platform_dataset"] = analyis_sources["platformDataSet"]["info"]
+
+    # changes the format of the annotation sources from Alissa, to fit in MongoDB and resemble the old O-schijf format
+    externalSources_dict = {}
+    variants = VUS_response
+    for key in analyis_sources['annotationSources']:
+        source_name = ""
+        source_value = ""
+        for x,y in key.items():
```

Comment:
I don't like the names `x, y`. If you need a non specific name, use `key, value`.

Reply:
yes, but key was already in use. but I agree with you statement. I might need to find a more non specific name

Reply:
changed first key and applied your suggestion 👍